File: baekjoon-online-judge/easy/02910.py
from collections import Counter
from sys import stdin

input = stdin.readline

if __name__ == '__main__':
    # Input
    N, C = map(int, input().split())
    A = [*map(int, input().split())]

    # Solve
    counter_dict_a = [*Counter(A).items()]
    counter_dict_a = [(idx, *item) for idx, item in enumerate(counter_dict_a)]
    counter_dict_a.sort(key=lambda x: (-x[2], x[0]))
    res = []
    for idx, key, value in counter_dict_a:
        for _ in range(value):
            res.append(key)

    # Output
    print(*res)

File: baekjoon-online-judge/easy/02910.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        // Init
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        // Input
        br.readLine();

        // Solve
        HashMap<Integer, Integer> hashMap = new LinkedHashMap<>();
        int key;
        for (String num : br.readLine().split(" ")) {
            key = Integer.parseInt(num);
            if (hashMap.containsKey(key)) {
                hashMap.replace(key, hashMap.get(key) + 1);
            } else {
                hashMap.put(key, 1);
            }
        }

        ArrayList<Integer> res = new ArrayList<>(hashMap.keySet());
        res.sort((o1, o2) -> Integer.compare(hashMap.get(o2), hashMap.get(o1)));

        // Output
        for (Integer cnt : res) {
            for (int i = 0; i < hashMap.get(cnt); ++i)
                sb.append(cnt).append(" ");
        }
        System.out.println(sb);

    }
}
